rdkit/Dbase/DbResultSet.py

```diff
@@ -60,52 +60,34 @@ def GetColumnTypes(self):
     return self.colTypes
 
   def GetColumnNamesAndTypes(self):
-    res = [None] * len(self.colNames)
-    for i in range(len(self.colNames)):
-      res[i] = self.colNames[i], self.colTypes[i]
-    return tuple(res)
+    return tuple(nt for nt in zip(self.colNames, self.colTypes))
 
 
 class DbResultSet(DbResultBase):
-  """ Only supports forward iteration
-
-  """
+  """ Only supports forward iteration """
 
   def __init__(self, *args, **kwargs):
     DbResultBase.__init__(self, *args, **kwargs)
-    self.seen = []
-    self._stopped = 0
 
-  def Reset(self):
-    self._stopped = 0
-    DbResultBase.Reset(self)
-
-  def next(self):
-    if self._stopped:
-      raise StopIteration
-    r = None
-    while r is None:
+  def __iter__(self):
```

Comment:
Though I agree that supporting the iterator interface is a good idea and should stay in the code, you also breaks backwards compatibility by removing the old interface. I don't think that is such a good idea, at least not without a longer discussion. Please put the old methods back.